### PR TITLE
Avoid boxing while unpacking flags

### PR DESCRIPTION
**Pull request by drewnoakes**
## Proposed changes

Eliminate boxing when in `GitItemStatus` bool property getters.

`Enum.HasFlag` boxes its argument. Replace it with a method that does not.

## Test methodology 

- Manual testing

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/Git/GitItemStatus.cs [hunk 1]
```diff
@@ -159,10 +159,16 @@ public bool IsStatusOnly
...
-            get => _flags.HasFlag(Flags.IsRangeDiff);
+            get => HasFlag(Flags.IsRangeDiff);
             set => SetFlag(value, Flags.IsRangeDiff);
         }
 
+        private bool HasFlag(Flags flags)
+        {
+            // NOTE Enum.HasFlag boxes its argument
+            return (flags & _flags) == flags;
+        }
```
Comment (mstv) [inline GitCommands/Git/GitItemStatus.cs:170]: BTW: The name of the enum `Flags` is misleading. It should be in singular.
I can read this better:
```suggestion
        private bool HasFlag(Flags flag)
        {
            // NOTE Enum.HasFlag boxes its argument
            return (_flags & flag) == flag;
        }
```

**Review by gerhardol — COMMENTED** [review]
+1
Agree with mstv

File: GitCommands/Git/GitItemStatus.cs (hunk 1 above)
Comment (drewnoakes) [reply GitCommands/Git/GitItemStatus.cs:170]: The logic works equally well with multiple flags. The type is a flags enum, and has a plural name.

**Review by RussKie — COMMENTED** [review]
IIRC .NET Core has resolved this allocations, hasn't it? #8522 makes this change obsolete.

**Comment by drewnoakes** [comment]
@RussKie what change are you looking for here?

**Review by gerhardol — APPROVED** [review]

**Comment by gerhardol** [comment]
> @RussKie what change are you looking for here?

I believe it was a comment that the change is not necessary as .NET5 should be coming soon
Suggest this is merged now, an improvement regardless

File: GitCommands/Git/GitItemStatus.cs (hunk 1 above)
Comment (mstv) [reply GitCommands/Git/GitItemStatus.cs:170]: > The logic works equally well with multiple flags.

This was clear to me. But `Enum.HasFlag` uses only singular. And nothing else is used here.

Better readability (as in other recent PRs):
```suggestion
        private bool HasFlag(Flags flags)
        {
            // NOTE Enum.HasFlag boxes its argument
            return (_flags & flags) == flags;
        }
```

**Review by RussKie — APPROVED** [review]

**Pull request merged**